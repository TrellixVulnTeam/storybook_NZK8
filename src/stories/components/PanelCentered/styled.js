// @flow
/* eslint-disable import/prefer-default-export */

import styled from 'styled-components';
import { colors } from 'utils/StyleGuide';

export const Wrapper = styled.div`
  position: relative;
  margin: 30px auto 20px auto;
  background-color: ${colors.white};
  border: 1px solid ${colors.paleGrey};
  border-radius: 8px;
  padding: 20px;
  width: 100%;

  /* Width only matters in desktop mode */
  @media (min-width: 1200px) {
    padding: 40px 20px;
    width: ${p => (p.width ? `${p.width}px` : '100%')};
  }
`;
