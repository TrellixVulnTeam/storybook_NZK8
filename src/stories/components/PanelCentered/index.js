// @flow

import React from 'react';
import { Wrapper } from './styled';

type Props = {
  width?: number,
  children: React$Node,
};

export default (props: Props) => <Wrapper width={props.width}>{props.children}</Wrapper>;
