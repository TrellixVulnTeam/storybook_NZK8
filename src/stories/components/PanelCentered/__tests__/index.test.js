import React from 'react';
import { shallow } from 'enzyme';
import PanelCentered from '../index';

describe('<PanelCentered />', () => {
  it('snapshot', () => {
    const c = shallow(<PanelCentered />);

    expect(c).toMatchSnapshot();

    c.setProps({
      width: 500,
    });

    expect(c).toMatchSnapshot();
  });

  it('should render props.children', () => {
    const c = shallow(
      <PanelCentered>
        <div id="foo" />
      </PanelCentered>,
    );

    expect(c.contains(<div id="foo" />)).toStrictEqual(true);
  });
});
