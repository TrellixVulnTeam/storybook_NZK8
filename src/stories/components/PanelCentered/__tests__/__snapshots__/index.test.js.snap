// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PanelCentered /> snapshot 1`] = `<styled__Wrapper />`;

exports[`<PanelCentered /> snapshot 2`] = `
<styled__Wrapper
  width={500}
/>
`;
